### ISSUE TITLE
Extractor: false positives on one-line/heredoc-embedded spoke invocations; gate-tool check floods on non-prompt inputs

### ISSUE THREAD

**Issue by belarusian** [belarusian/spoke-lint#60]
## Repro

```
cd ~/AI/spoke-lint/proj
python3 -m spoke_lint.cli check ~/AI/launch-gate/launch-setup.sh --spokes-dir ~/Research/four/examples/spokes
# exit 1
```

Input is a mission-compiler-generated launch script (shell, not a runner prompt) whose only spoke invocation is heredoc-embedded as ONE long line:

```
INNER=$(cat <<'INNER_EOF'
python3 /home/sasha/Research/four/examples/spokes/project-setup.py --goal "Mission: Build launch-gate: ..." --name launch-gate --project-dir /home/sasha/AI/launch-gate/proj --ai-dir /home/sasha/AI/launch-gate/ai --cycles 12 --repo belarusian/launch-gate --seed /home/sasha/AI/launch-gate/seed
INNER_EOF
)
```

### 1. Extractor gap: one-line invocation with a long quoted value mis-parsed

Reported:
```
unknown_flag: m — Unknown flag --m passed to .../project-setup.py
missing_required: name / project-dir / ai-dir
```

But the real argparse (verified via `project-setup.py --help`) accepts exactly the 7 flags present on that line (`--goal --name --project-dir --ai-dir --cycles --repo --seed`); all are there. The extractor stops/garbles inside the long quoted `--goal "..."` value and loses the trailing flags, inventing `--m`.

This is not only a launch-script dialect: the setup spoke generates the runner prompt with the same one-line invocation form, so spoke-lint will false-positive on the real artifact too.

### 2. Gate-tool check floods on non-prompt inputs

~40 `missing_tool` lines for `set`, `export`, `PROJECT_DIR="..."`, `<<'GOAL_EOF'`, `Mission:`, `-`, `)` — the first token of every shell line is treated as a gate command. The tool's input contract is runner prompts (markdown + bash blocks); a shell launch script is not a valid input, and the failure mode should be a clean exit-2 "not a runner prompt", not a false-positive flood that buries any real findings.

## Suggested fix

1. Extractor: handle one-line invocations whose quoted values contain spaces/collos — tokenize respecting quote boundaries to end-of-line, not first whitespace.
2. CLI: detect non-runner-prompt inputs (no markdown/bash-block structure, or pure shell) and exit 2 with a single diagnostic line.

## Evidence on Sunny

- Input file: `~/AI/launch-gate/launch-setup.sh`
- Full output captured 2026-08-22 during launch-gate T6 pre-flight (manual argparse verification passed; launch proceeded on that basis).

**Comment by belarusian** [comment]
Delivered in PR #67 (merged as commit a7a6aa5).

- **Quote-aware tokenizer** (`spoke_lint/extractor.py`): `_tokenize` now respects quote boundaries and consumes an unterminated quote to end-of-line, so a long `--goal "..."` value no longer swallows trailing flags or invents a spurious `--m` flag from the `-m` inside the quoted text.
- **Non-runner-prompt input contract** (`spoke_lint/contract.py` + `spoke_lint/cli.py`): new pure `is_runner_prompt` predicate detects pure shell launch scripts (shebang / heredoc / ≥2 shell-statement lines outside fenced blocks); the CLI rejects them with exit 2 and a single clean diagnostic instead of the ~40-line `missing_tool` flood. Reuses the Cycle 14 markdown fence-awareness.

Gate green on merge: 212 passed, ruff clean, mypy clean (10 files). Repro (`check ~/AI/launch-gate/launch-setup.sh`) now exits 2 with one diagnostic line.
